tests/bluetooth/ctrl_isoal/src/main.c

```diff
@@ -20,1517 +20,7747 @@
-	/* Expecting 0 data to be written, with no error status */
-	zassert_equal(sink->sdu_production.sdu_written, 0, "written=%u",
-		      sink->sdu_production.sdu_written);
-	zassert_equal(sink->sdu_production.sdu_status, ISOAL_SDU_STATUS_ERRORS, "sdu_status=0x%x",
-		      sink->sdu_production.sdu_status);
-
-	/* Third and last padding in SDU interval */
-	sdu_emit_expected = false;
-	construct_pdu_unframed(PDU_BIS_LLID_START_CONTINUE, pdu_ref, 0 /* len = 0 => padding */,
-			       payload_number++, 923751, 0);
-	err = isoal_rx_pdu_recombine(sink_hdl, &pdu_meta);
-	zassert_equal(err, ISOAL_STATUS_OK, "err=0x%02x", err);
-	/* Expecting 0 data to be written but with error status */
-	zassert_equal(sink->sdu_production.sdu_written, 0, "written=%u",
-		      sink->sdu_production.sdu_written);
-	zassert_equal(sink->sdu_production.sdu_status, ISOAL_SDU_STATUS_ERRORS, "sdu_status=0x%x",
-		      sink->sdu_production.sdu_status);
+	/* Test recombine (Black Box) */
+	/* Should not allocate a new SDU */
+
+	/* SDU should not be written to */
+
+	/* SDU should not be emitted */
+
+	err = isoal_rx_pdu_recombine(sink_hdl, &rx_pdu_meta_buf.pdu_meta);
+
+	zassert_equal(err, ISOAL_STATUS_OK, "err=0x%02x", err);
 }
 
 /**
- * @brief   Padding PDU with error status received in the ISOAL_CONTINUE state (middle of SDU)
+ * Test Suite	:	RX unframed PDU reassembly
+ *
+ * Tests reassembly of a single valid RX PDU into an SDU.
  */
-void test_unframed_padding_error2(void)
+ZTEST(test_rx_unframed, test_rx_unframed_single_pdu)
 {
-	clear_sdu_buf();
-
-	/* Assumes 3 PDUs per SDU interval */
+	struct rx_pdu_meta_buffer rx_pdu_meta_buf;
+	struct rx_sdu_frag_buffer rx_sdu_frag_buf;
+	struct isoal_sdu_buffer sdu_buffer;
+	isoal_sink_handle_t sink_hdl;
+	uint32_t stream_sync_delay;
+	uint32_t group_sync_delay;
+	isoal_sdu_len_t sdu_size;
+	uint8_t iso_interval_int;
+	uint64_t payload_number;
+	uint32_t pdu_timestamp;
+	uint32_t sdu_timestamp;
+	uint16_t testdata_indx;
+	uint16_t testdata_size;
+	uint32_t sdu_interval;
+	isoal_sdu_cnt_t seqn;
+	uint8_t testdata[23];
+	isoal_status_t err;
+	uint32_t latency;
+	uint8_t role;
+	uint8_t BN;
+	uint8_t FT;
+
+	/* Settings */
+	role = BT_CONN_ROLE_PERIPHERAL;
+	iso_interval_int = 1;
+	sdu_interval = CONN_INT_UNIT_US;
+	BN = 1;
+	FT = 1;
+	stream_sync_delay = (iso_interval_int * CONN_INT_UNIT_US) - 200;
+	group_sync_delay = (iso_interval_int * CONN_INT_UNIT_US) - 50;
+
+	/* PDU 1 -------------------------------------------------------------*/
+	init_rx_pdu_buffer(&rx_pdu_meta_buf);
+	init_rx_sdu_buffer(&rx_sdu_frag_buf);
+	init_test_data_buffer(testdata, 23);
+	sdu_buffer.dbuf = &rx_sdu_frag_buf;
+	sdu_buffer.size = TEST_RX_SDU_FRAG_PAYLOAD_MAX;
+	latency =
+	payload_number = 2000;
+	pdu_timestamp = 9249;
+	latency = calc_rx_latency_by_role(role, false, FT,
+					sdu_interval, iso_interval_int,
+					stream_sync_delay, group_sync_delay);
+	sdu_timestamp = (uint32_t)((int64_t)pdu_timestamp + latency);
```

Comment:
```suggestion
	sdu_timestamp = (uint32_t)((uint64_t)pdu_timestamp + latency);
```

What is the point of the 64-bit cast if we are also casting the value to a 32-bit value afterwards?

Reply:
I did have some concerns as to how a sum of an unsigned and signed variables would be handled in the unit test. I don't recall now if that caused a problem but maybe it was unfounded.

Reply:
If you are risking that `(uint32_t) + (uint32_t) > UINT32_MAX` then it makes sense to cast to `uint64_t`, but it doesn't really makes sense to then cast the `uint64_t` to a `uint32_t` (and certainly doesn't make sense to cast the addition of two unsigned to a signed value first).

Reply:
Let me consider that again. It's been some months and I don't remember exactly what I was trying to cross-check. :-)